variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/repositories/VariantRepositoryImpl.java

```diff
@@ -108,6 +109,32 @@ public Long countByRegionsAndComplexFilters(List<Region> regions, List<VariantRe
                 .distinct(VariantMongo.CHROMOSOME_FIELD));
     }
 
+    @Override
+    public List<VariantMongo> findByRegionAndOtherBeaconFilters(Region startRange, Region endRange,
+                                                                List<VariantRepositoryFilter> filters) {
+        Query query = new Query();
+        query.addCriteria(Criteria.where(VariantMongo.CHROMOSOME_FIELD).is(startRange.getChromosome()));
+
+        Criteria startCriteria = getRegionRangeCriteria(startRange, VariantMongo.START_FIELD);
+        Criteria endCriteria = getRegionRangeCriteria(endRange, VariantMongo.END_FIELD);
+        if (startCriteria != null) {
+            query.addCriteria(startCriteria);
+        }
+
+        if (endCriteria != null) {
+            query.addCriteria(endCriteria);
+        }
+        Pageable pageable = new PageRequest(0,(int)getVariantMongoTotalCountHelper());
+        List<VariantMongo> variantMongoList = findByComplexFiltersHelper(query, filters, null, pageable);
+        return variantMongoList;
+    }
+
+    private long getVariantMongoTotalCountHelper() {
+        Aggregation aggregation = Aggregation.newAggregation(Aggregation.group().count().as("count"));
+        AggregationResults<VariantAggregationCount> aggregationResults= mongoTemplate.aggregate(aggregation,VariantMongo.class,VariantAggregationCount.class);
```

Comment:
no need for an aggregation, this whole method can be replaced by `mongoTemplate.count(query, VariantMongo.class)`. Also, with this aggregation you are counting the total documents in the table, not with the relevant query

Reply:
Okay, I will change it :)
I wrote the aggregation to count the total number of documents since that can act as the maximum number of documents that it can return.

Reply:
Fixed :+1: